src/genpy/generator.py

```diff
@@ -896,6 +897,7 @@ def srv_generator(msg_context, spec, search_path):
     genmsg.msg_loader.load_depends(msg_context, spec, search_path)
     md5 = genmsg.compute_md5(msg_context, spec)
 
+    yield '# This Python file uses the following encoding: utf-8'
```

Comment:
Actually this function doesn't need the change since it generates the message beforehand.